crates/sui-bridge-indexer/src/eth_bridge_indexer.rs

```diff
@@ -175,87 +183,151 @@ impl Datasource<RawEthData> for EthFinalizedSyncDatasource {
                 .interval(std::time::Duration::from_millis(2000)),
         );
 
-        let bridge_address = self.bridge_address;
+        let bridge_addresses = self.bridge_addresses.clone();
         let indexer_metrics: BridgeIndexerMetrics = self.indexer_metrics.clone();
         let client = Arc::clone(&client);
         let provider = Arc::clone(&provider);
+        let bridge_metrics = Arc::clone(&self.bridge_metrics);
+        let current_block = provider.get_block_number().await?.as_u64();
 
         let handle = spawn_monitored_task!(async move {
-            let mut cached_blocks: HashMap<u64, Block<H256>> = HashMap::new();
-
-            let mut last_finalized_block = retry_with_max_elapsed_time!(
-                client.get_last_finalized_block_id(),
-                Duration::from_secs(30000)
-            )
-            .expect("Unable to get latest finalized block from provider")
-            .expect("Failed to unwrap the inner result");
-
-            // block process until current range is finalized
-            while target_checkpoint > last_finalized_block {
-                tokio::time::sleep(Duration::from_secs(5)).await;
-
-                last_finalized_block = retry_with_max_elapsed_time!(
-                    client.get_last_finalized_block_id(),
-                    Duration::from_secs(30000)
+            if target_checkpoint > current_block {
```

Comment:
this feel a bit unsafe. and on sui side there are a few places like this which i don't quite like. Can we add a TODO here and we will come back with a consistent and robust alternative